scipy/stats/stats.py

```diff
@@ -1088,14 +1091,36 @@ def variation(a, axis=0):
         Axis along which to calculate the coefficient of variation. Default
         is 0. If None, compute over the whole array `a`.
 
+    Returns
+    -------
+    variation : float or array
+        Coefficient ot variation as defined by `s / m` where `s` is the
+        standard deviation and `m` is the mean.
+
+    Notes
+    -----
+    Calculating the coefficient of variation only makes sense for non-symmetric
+    distributions.
+
     References
     ----------
     .. [1] Zwillinger, D. and Kokoska, S. (2000). CRC Standard
        Probability and Statistics Tables and Formulae. Chapman & Hall: New
        York. 2000.
 
+    Examples
+    --------
+    >>> from scipy import stats
+    >>> a = np.linspace(stats.erlang.ppf(0.01, 1), 100)
+    >>> stats.variation(a)
+    0.58889670528253069
+
     """
     a, axis = _chk_asarray(a, axis)
+
+    if a.size == 0:
+        return np.nan
```

Comment:
The logic is here: https://github.com/numpy/numpy/blob/master/numpy/core/src/umath/reduction.c#L194

I'm not aware of a public function in numpy that provides the right shape (I could be missing it - anyone?), but it's not too difficult to reimplement here in a utility function.